rfcs/2021-19-02-units-for-typography-and-layout.md

```diff
@@ -0,0 +1,603 @@
+| Tyography auto scale, layout pixel ratio(native) |           ✅           |          ❌           |         ✅         |        🚫         |
+
+>⚠️ Doesn't breaks but doesn't renders as expected
+### Accessibility Guideline?
+The WCAG guideline says that content satisfies the **success criterion** if it can be **scaled up to 200%**. It can go beyond that but logically beyond that the scaling becomes more extreme and adaptive layouts may introduce usability problems.
+
+For example, words may be too wide to fit into the horizontal space available to them, causing them to be truncated; layout constraints may cause text to overlap with other content when it is scaled larger; or only one word of a sentence may fit on each line, causing the sentence to be displayed as a vertical column of text that is difficult to read.
+>🔗 [WCAG guideline link](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html)
+### What units other Design Systems are using?
+
+| Name                                                                                     | Typography | Layout | Comments                                 |
+| ---------------------------------------------------------------------------------------- | ---------- | ------ | ---------------------------------------- |
+| [Primer - GitHub](https://primer.style/css/support/spacing)                              | px         | px     | NA                                       |
+| [Spectrum - Adobe](https://react-spectrum.adobe.com/react-spectrum/getting-started.html) | px         | px     | revisitng decision                       |
+| [Braid - Seek](https://seek-oss.github.io/braid-design-system/)                          | px         | px     | revisiting decision                      |
+| [Styled System](https://styled-system.com/)                                              | px         | px     | default px, can change by suffixing unit |
+| [Facebook](https://twitter.com/naman34/status/1362626358539481090)                       | rems       | px     | NA                                       |
+| [Chakra](https://chakra-ui.com/docs/getting-started)                                     | rems       | rems   | default rems, option for other units     |
+
+>📝 Note: Couldn't find anything similar for react native
+### What will work for us?
+Looking at the POC and the [matrix](#absolute-unit-vs-relative-unit-matrix), relative units looks no brainer but it has some downsides too:
+* Assume we have few responsive layouts then increasing the font-size on desktop might trigger the media queries to kick in and because of that our layout will be re-arranged assuming that the screen size has changed. Now the user was just intending to increase the font-size but they got tricked into different layout since everything is relative which impacts their experience.
+* Since everything is relative, increase in font-size is identical to page zoom so the intent of the user to increase the content size goes for a toss.
```

Comment:
Isn't this fine? Even for a page zoom, if there isn't enough room in the UI to layout as a "dektop" design the UI will update to a "mobile" design?

Reply:
so assume if your intent was just the content zoom but not the layout zoom so this defeats that as by zooming in you are just scaling the layout(it's like magnifier). 

On a side note page zoom doesn't triggers media queries it just scales things and the rendering engine doesn't kicks in at all.